app/chat/tasks.py

```diff
@@ -0,0 +1,119 @@
+from app.redis_service import RedisService
+from celery import app, group
+from celery.utils.log import get_task_logger
+from django.conf import settings
+from mattermostdriver import Driver
+
+logger = get_task_logger(__name__)
+
+redis = RedisService().redis
+
+# Lock timeout of 2 minutes (just in the case that the application hangs to avoid a redis deadlock)
+LOCK_TIMEOUT = 60 * 2
+
+chat_driver = Driver({
+    'url': settings.CHAT_URL,
+    'port': settings.CHAT_PORT,
+    'token': settings.CHAT_DRIVER_TOKEN
+})
+
+
+@app.shared_task(bind=True, max_retries=3)
+def create_channel(self, options, retry: bool = True) -> None:
+    """
+    :param options:
+    :param retry:
+    :return:
+    """
+    with redis.lock("tasks:create_channel:%s" % options['channel_name'], timeout=LOCK_TIMEOUT):
```

Comment:
Why the need for locks here?

Reply:
if 2 hunters requested to start work before a channel has been created, 2 messages could get queued to create the same channel, but if we lock with redis we dont have to worry about potentially trying to create 2.